validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/SlashingProtectedValidatorSource.java

```diff
@@ -52,21 +53,25 @@ public DeleteKeyResult deleteValidator(final BLSPublicKey publicKey) {
   }
 
   @Override
-  public AddLocalValidatorResult addValidator(
+  public AddValidatorResult addValidator(
       final KeyStoreData keyStoreData, final String password, final BLSPublicKey publicKey) {
-    AddLocalValidatorResult delegateResult =
-        delegate.addValidator(keyStoreData, password, publicKey);
+    AddValidatorResult delegateResult = delegate.addValidator(keyStoreData, password, publicKey);
 
     if (delegateResult.getSigner().isEmpty()) {
       return delegateResult;
     }
 
     final Signer signer = delegateResult.getSigner().get();
-    return new AddLocalValidatorResult(
+    return new AddValidatorResult(
         delegateResult.getResult(),
         Optional.of(new SlashingProtectedSigner(publicKey, slashingProtector, signer)));
   }
 
+  @Override
+  public AddValidatorResult addValidator(BLSPublicKey publicKey, URL signerUrl) {
+    throw new UnsupportedOperationException();
+  }
```

Comment:
I think this will be where we wrap in a `SlashingProtectedSigner` is basically the same way we do for `addValidator`.  Then just need to make sure we wrap the `ExternalValidatorSource` with a. `SlashingProtectedValidatorSource` when we create it.

Reply:
We might need to dig into the design, there was a `mutableValidatorSource` which is basically a non read-only local validator source... we'll probably add the writable concept through `ExternalValidatorSource`, and have an instance of `ExternalValidatorSource` that's not read-only...

Reply:
Let's catch up and I can fill you in on discussions before we give Courtney too much conflicting advice. :) There's definitely refactoring opportunities we've not taken so that we can get to something that works and know we've understood all the key interactions.